wavelength-plugins/profile.js

```diff
@@ -282,32 +287,119 @@ exports.commands = {
 		},
 	},
 
+	favoritetype: 'type',
+	type: {
+		add: "set",
+		set: function (target, room, user) {
+			if (!target) return this.parse("/help type");
+			let type = Dex.getType(target);
+			if (!type.exists) return this.errorReply('Not a type. Check your spelling?');
+			Db.type.set(user.userid, toId(type));
+			return this.sendReply(`Your favorite type has been set to "${type}".`);
+		},
+
+		del: "delete",
+		remove: "delete",
+		delete: function (target, room, user) {
+			if (!Db.type.has(user.userid)) return this.errorReply("Your Favorite Type hasn't been set.");
+			Db.type.remove(user.userid);
+			return this.sendReply("Your favorite type has been deleted from your profile.");
+		},
+
+		"": "help",
+		help: function (target, room, user) {
+			this.parse('/help type');
+		},
+	},
+	typehelp: [
+		"/type set [type] - Sets your Favorite Type.",
+		"/type delete - Removes your Favorite Type.",
+	],
+
+	profilecolor: 'pcolor',
+	pcolor: {
+		set: 'add',
+		add: function (target, room, user) {
+			if (!target) return this.parse('/pcolor help');
+			let color = target.trim();
+			if (color.charAt(0) !== '#') return this.errorReply(`The color needs to be a hex starting with "#".`);
+			Db.profilecolor.set(user, color);
+			this.sendReply(`You have set your profile color to "${color}".`);
+		},
+
+		// For staff
+		forceset: 'forceadd',
+		forceadd: function (target, room, user) {
```

Comment:
Not really sure why this command is needed.

Reply:
Even if it is, users can just change it back.

Reply:
Feel free to remove it. I thought it could be useful if an user set his/her profile color to something that's not readable and refuses to change, staff can change the color themself.